File: src/main/java/com/example/board/model/Post.java
package src.main.java.com.example.board.model;

import java.time.Clock;
import java.time.LocalDateTime;

public class Post {

    private final Long id;
    private final String title;
    private final String content;
    private final String author;
    private final LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Post() {
        // Default constructor for frameworks like JPA
        this.id = null;
        this.title = null;
        this.content = null;
        this.author = null;
        this.createdAt = null;
        this.updatedAt = null;
    }

    public Post(Long id, String title, String content, String author, Clock clock) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.author = author;
        this.createdAt = LocalDateTime.now(clock);
        this.updatedAt = null;
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getContent() {
        return content;
    }

    public String getAuthor() {
        return author;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: app/models/post.py
from datetime import datetime

class Post:
    """
    게시글 도메인 모델 (정상 코드 — 대조군).
    """

    def __init__(self, id: int = None, title: str = None, content: str = None, author: str = None):
        self.id = id
        self.title = title
        self.content = content
        self.author = author
        self.created_at = datetime.now() if id and title and content and author else None

    def get_id(self) -> int:
        """게시글의 고유 ID를 반환합니다."""
        return self.id

    def get_title(self) -> str:
        """게시글의 제목을 반환합니다."""
        return self.title

    def get_content(self) -> str:
        """게시글의 내용을 반환합니다."""
        return self.content

    def get_author(self) -> str:
        """게시글의 저자를 반환합니다."""
        return self.author

    def get_created_at(self) -> datetime:
        """게시글이 생성된 날짜 및 시간을 반환합니다."""
        return self.created_at